### PR TITLE
ETCD-588: Add e2e test for hardware speed profiles feature

### PR DESCRIPTION
**Pull request by dusk125**

### CODE REVIEW

File: test/extended/etcd/hardware_speed.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/pkg/errors"
+
+	"github.com/openshift/library-go/test/library"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("[sig-etcd][Feature:HardwareSpeed][serial] etcd", func() {
```
Comment (soltysh) [inline test/extended/etcd/hardware_speed.go:19]:
```suggestion
var _ = g.Describe("[sig-etcd][Feature:HardwareSpeed][Serial] etcd", func() {
```

File: test/extended/etcd/hardware_speed.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	// next it validates that the etcd and api servers come back up and are healthy.
+	// next it sets the hardware speed to Slower.
+	// and validates that the etcd and api servers come back up and are healthy.
+	// The test ends by resetting the hardware speed to default.
+	// and validates that the etcd and api servers come back up and are healthy.
+	g.It("is able to update the hardware speed [Timeout:60m][apigroup:machine.openshift.io]", func() {
+		// Set the hardware speed to Standard from default ""
+		g.GinkgoT().Log("setting hardware speed to Standard")
+		data := fmt.Sprintf(`{"spec": {"controlPlaneHardwareSpeed": "Standard"}}`)
+		_, err := oc.AdminOperatorClient().OperatorV1().Etcds().Patch(context.Background(), "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
```
Comment (soltysh) [inline test/extended/etcd/hardware_speed.go:54]: All of the context invocations should be replaced with a context injected into `g.It` like so:
```
g.It("is able to update the hardware speed [Timeout:60m][apigroup:machine.openshift.io]", func(ctx context.Context) {
```

File: test/extended/etcd/hardware_speed.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	})
+
+	// The following test covers a hardware speed change.
+	// It starts by changing the hardware speed to Standard.
+	// next it validates that the etcd and api servers come back up and are healthy.
+	// next it sets the hardware speed to Slower.
+	// and validates that the etcd and api servers come back up and are healthy.
+	// The test ends by resetting the hardware speed to default.
+	// and validates that the etcd and api servers come back up and are healthy.
+	g.It("is able to update the hardware speed [Timeout:60m][apigroup:machine.openshift.io]", func() {
```
Comment (soltysh) [inline test/extended/etcd/hardware_speed.go:50]: 60m? is it that slow?

**Comment by soltysh** [comment]
You're missing `make update`

File: test/extended/etcd/hardware_speed.go (hunk 3 above)
Comment (dusk125) [reply test/extended/etcd/hardware_speed.go]: It shouldn't be, I adjusted it to be the same timeout as the waits. It shouldn't take that long to stabilize.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

Got the desired signal:
```
started: 0/23/44 "[sig-etcd][Feature:HardwareSpeed][Serial] etcd is able to update the hardware speed [Timeout:30m][apigroup:machine.openshift.io] [Suite:openshift/conformance/serial]"

passed: (10m9s) 2024-04-25T11:27:09 "[sig-etcd][Feature:HardwareSpeed][Serial] etcd is able to update the hardware speed [Timeout:30m][apigroup:machine.openshift.io] [Suite:openshift/conformance/serial]"
```

**Pull request merged**